sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -320,74 +356,39 @@ private static int speculateContentLength(HttpHeaders headers) {
-            }
-        }
 
         /**
          * Calls buildAndSend to send a String representation of the request across the output stream, then calls
          * buildResponse to get an instance of HttpUrlConnectionResponse from the input stream
          *
          * @param httpRequest The HTTP Request being sent
-         * @param socket An instance of the SocketClient
+         * @param bufferedInputStream the input stream from the socket
+         * @param outputStream the output stream from the socket for writing the request
          * @return an instance of Response
          */
-        @SuppressWarnings("deprecation")
-        private static Response<?> doInputOutput(HttpRequest httpRequest, Socket socket) throws IOException {
+        private static Response<?> sendPatchRequest(HttpRequest httpRequest, BufferedInputStream bufferedInputStream,
+            OutputStream outputStream) throws IOException {
             httpRequest.getHeaders().set(HttpHeaderName.HOST, httpRequest.getUrl().getHost());
-            if (!"keep-alive".equalsIgnoreCase(httpRequest.getHeaders().getValue(HttpHeaderName.CONNECTION))) {
-                httpRequest.getHeaders().set(HttpHeaderName.CONNECTION, "close");
-            }
-
-            try (BufferedReader in = new BufferedReader(
-                new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8));
-                OutputStreamWriter out = new OutputStreamWriter(socket.getOutputStream())) {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(bufferedInputStream, StandardCharsets.UTF_8));
+            OutputStreamWriter out = new OutputStreamWriter(outputStream);
 
-                buildAndSend(httpRequest, out);
+            buildAndSend(httpRequest, out);
+            Response<?> response = buildResponse(httpRequest, reader);
 
-                Response<?> response = buildResponse(httpRequest, in);
-                HttpHeader locationHeader = response.getHeaders().get(HttpHeaderName.LOCATION);
-                String redirectLocation = (locationHeader == null) ? null : locationHeader.getValue();
+            HttpHeader locationHeader = response.getHeaders().get(HttpHeaderName.LOCATION);
+            String redirectLocation = (locationHeader == null) ? null : locationHeader.getValue();
 
-                if (redirectLocation != null) {
-                    if (redirectLocation.startsWith("http")) {
-                        httpRequest.setUrl(redirectLocation);
-                    } else {
-                        httpRequest.setUrl(new URL(httpRequest.getUrl(), redirectLocation));
-                    }
-                    return sendPatchRequest(httpRequest);
+            if (redirectLocation != null) {
+                if (redirectLocation.startsWith("http")) {
+                    httpRequest.setUrl(redirectLocation);
+                } else {
+                    httpRequest.setUrl(new URL(httpRequest.getUrl(), redirectLocation));
```

Comment:
Should make this check more durable, what if we get back a full URL without the leading protocol, or what if the redirect location plus original URL don't make a valid URL

Reply:
https://github.com/orgs/Azure/projects/579/views/2?pane=issue&itemId=61027721